services/sessions/preview.ts

```diff
@@ -0,0 +1,40 @@
+import { ResponseCookie } from "next/dist/compiled/@edge-runtime/cookies";
+import { cookies, draftMode } from "next/headers";
+
+const previewSession = () => {
+  const cookieSettings: Partial<ResponseCookie> = {
+    httpOnly: true,
+    sameSite: true,
+    secure: true,
+  };
+  const tokenCookie = "previewToken";
+
+  const draft = () => {
```

Comment:
wrapping this inside of try-catch because if the client is used for `generateStaticParams` then draft mode is not available and throws an error. 

Alternatively a separate client could be created that doesn't include the preview mode checks, but quashing the error seems fine to me.